### DISCUSSION TITLE
Run serverpod generate in modules and core, too.

### DISCUSSION THREAD

**Discussion by fischerscode** [serverpod/serverpod#653] [Ideas]
There are use cases, where it would be useful, if it was possible to generate module stuff based on the configuration of the developers server package. (One example: Add additional, nullable column to a Table)

This night, I had an idea, how this might possible...
1. When `serverpod generate` runs, copy the dart packages of the relevant modules to `.dart_tool`
2. Run the generator inside `.dart_tool` for each module. The generator is now able to parse the main package, too.
3. Modify the `pubspec.lock`s to link the packages manually.

So basically the definition of a module could depend on the developers project.

This is just an idea, I haven't done any research / tests. Just wanted to document it and get your thoughts about this.

**Comment by wstrange** [comment]
To create the tables needed by the auth module, I had to copy/paste them out of github.  Am I doing it wrong?

Is there some way to get `serverpod generate` to add them to my `tables-serverpod.pgsql` automagically?

Given how popular the auth module is, would it makes sense to include this out of the box.  Worst case it's just a few tables that might not get used.

**Reply by fischerscode** [reply]
Something like the approach I mentioned could solve this. For now, I don't think it's a good idea to add the tables by default, since it might mess with the classes the developer defines.

> Is there some way to get serverpod generate to add them to my tables-serverpod.pgsql automagically?

You could merge all files (serverpod, modules and yours) into one using GitHub Actions.
